graph/pull_v2.go

```diff
@@ -225,12 +224,13 @@ func (p *v2Puller) pullV2Tag(out io.Writer, tag, taggedName string) (verified bo
 	}()
 	defer func() {
 		if err != nil {
-			// All operations on the pipe are synchronous. This call will wait
-			// until all current readers/writers are done using the pipe then
-			// set the error. All successive reads/writes will return with this
-			// error.
-			pipeWriter.CloseWithError(errors.New("download canceled"))
+			err = fmt.Errorf("download canceled: %v", err)
```

Comment:
I don't think we want to do this. It will mean any error condition causes `pullV2Tag` to return "download canceled" instead of the original error.